lib/ios-deploy.js

```diff
@@ -64,23 +64,25 @@ class IOSDeploy {
     const afcService = await services.startAfcService(this.udid);
     try {
       const bundlePathOnPhone = await this.createAppPath(afcService, app);
-      const promises = [];
       await fs.walkDir(app, true, async (itemPath, isDir) => {
         const pathOnPhone = path.join(bundlePathOnPhone, path.relative(app, itemPath));
         if (isDir) {
           await afcService.createDirectory(pathOnPhone);
         } else {
-          const readStream = fs.createReadStream(itemPath, {autoClose: true});
-          const writeStream = await afcService.createWriteStream(pathOnPhone, {autoDestroy: true });
-          promises.push(new B((resolve) => writeStream.on('close', resolve)));
+          const readStream = fs.createReadStream(itemPath, { autoClose: true });
+          const writeStream = await afcService.createWriteStream(pathOnPhone, { autoDestroy: true });
+          const filePush = new B((resolve, error) => {
+            writeStream.on('close', resolve);
+            writeStream.on('error', error);
+          });
           readStream.pipe(writeStream);
+          try {
+            await filePush;
+          } catch (e) {
+            throw new Error(`Couldn't push the file '${itemPath}' within the timeout ${APPLICATION_PUSH_TIMEOUT}ms. Error: ${e.message}`);
+          }
         }
       });
-      try {
-        await B.all(promises).timeout(APPLICATION_PUSH_TIMEOUT);
```

Comment:
the problem might be, that we were trying to push all files at once. What if we push them chunk by chunk though?